lib/hanami_context_logging/logger.rb

```diff
@@ -15,13 +15,16 @@ class Logger < Hanami::Logger
     # @param app_name [String] app name (defaults to app name)
     # @param *args [any] any arbitrary Logger argument (will just be passed to Hanami::Logger)
     # @param options [Hash] Any Hanami::Logger options
-    def initialize(app_name = default_application_name, *args, options) # rubocop:disable Style/OptionalArguments
-      @initializing_arguments = [app_name, args, options]
+    def initialize(application_name = nil, *args, **kwargs) # rubocop:disable Layout/LineLength
+      # rubocop:disable Style/OptionalArguments
+      @initializing_arguments = [application_name, args, kwargs]
+
+      options_copy = kwargs.dup
+      formatter ||= :with_context # default formatter
```

Comment:
I have a question here, the formatter looks unused (maybe I am not familiar with the syntax). I also see another similar operation in line 26. Can I check what is this line achieving?

Reply:
You're right, this is just a me wrongly resolving conflict with stashed changes. However, for `context_provider` it is correct, as we don't want to pass it through to the `super`, while we still want to use it later.